ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+
+    for (i = 0, camp = bd_xp->pField; i < n_camps; i++, camp++)
+    {
+        MM_InitializeField(camp);
+        if (i < 99999)
+            sprintf(camp->FieldName, "CAMP%05u", (unsigned)(i + 1));
+        else
+            sprintf(camp->FieldName, "CM%u", (unsigned)(i + 1));
+        camp->FieldType = 'C';
+        camp->DecimalsIfFloat = 0;
+        camp->BytesPerField = 50;
+    }
+    return bd_xp;
+}
+
+MM_BYTE MM_DBFFieldTypeToVariableProcessing(MM_BYTE tipus_camp_DBF)
+{
+    switch (tipus_camp_DBF)
+    {
+        case 'N':
+            return MM_QUANTITATIVE_CONTINUOUS_FIELD;
+        case 'D':
+        case 'C':
+        case 'L':
+            return MM_CATEGORICAL_FIELD;
+    }
+    return MM_CATEGORICAL_FIELD;
+}
+
+static MM_BYTE MM_GetDefaultDesiredDBFFieldWidth(const struct MM_FIELD *camp)
+{
+    size_t a, b, c, d, e;
+
+    b = strlen(camp->FieldName);
+    c = strlen(camp->FieldDescription[0]);
+
+    if (camp->FieldType == 'D')
+    {
+        d = (b > c ? b : c);
+        a = (size_t)camp->BytesPerField + 2;
+        return (MM_BYTE)(a > d ? a : d);
+    }
+    a = camp->BytesPerField;
+    d = (unsigned int)(b > c ? b : c);
+    e = (a > d ? a : d);
+    return (MM_BYTE)(e < 80 ? e : 80);
+}
+
+static MM_BOOLEAN MM_is_field_name_lowercase(const char *cadena)
+{
+    const char *p;
+
+    for (p = cadena; *p; p++)
+    {
+        if ((*p >= 'a' && *p <= 'z'))
+            return TRUE;
+    }
+    return FALSE;
+}
+
+static MM_BOOLEAN
+MM_Is_classical_DBF_field_name_or_lowercase(const char *cadena)
+{
+    const char *p;
+
+    for (p = cadena; *p; p++)
+    {
+        if ((*p >= 'a' && *p <= 'z') || (*p >= 'A' && *p <= 'Z') ||
+            (*p >= '0' && *p <= '9') || *p == '_')
+            ;
+        else
+            return FALSE;
+    }
+    if (cadena[0] == '_')
+        return FALSE;
+    return TRUE;
+}
+
+static MM_BOOLEAN
+MM_Is_character_valid_for_extended_DBF_field_name(int valor,
+                                                  int *valor_substitut)
+{
+    if (valor_substitut)
+    {
+        switch (valor)
+        {
+            case 32:
+                *valor_substitut = '_';
+                return FALSE;
+            case 91:
+                *valor_substitut = '(';
+                return FALSE;
+            case 93:
+                *valor_substitut = ')';
+                return FALSE;
+            case 96:
+                *valor_substitut = '\'';
+                return FALSE;
+            case 127:
+                *valor_substitut = '_';
+                return FALSE;
+            case 168:
+                *valor_substitut = '-';
+                return FALSE;
+        }
+    }
+    else
+    {
+        if (valor < 32 || valor == 91 || valor == 93 || valor == 96 ||
+            valor == 127 || valor == 168)
+            return FALSE;
+    }
+    return TRUE;
+}
+
+static int MM_ISExtendedNameBD_XP(const char *nom_camp)
+{
+    GInt32 mida, j;
+
+    mida = (GInt32)strlen(nom_camp);
```

Comment:
I suspect the cast could be avoided with:

```suggestion
    size_t mida, j;

    mida = strlen(nom_camp);
```

Reply:
Sure! Done